torch/_inductor/lowering.py

```diff
@@ -4989,6 +4998,44 @@ def register_pointwise_numeric_ldf64(op):
 register_foreach_pointwise(aten._foreach_copy, copy)
 
 
+# these are only encountered as outputs of the graph
+# reinplacing epilogue copies improves compile time
+# by removing extra buffers sent to the scheduler.
+def register_foreach_inplace(aten_op, outplace_aten_op, outplace_op):
+    inplaceable_foreach_ops[outplace_aten_op] = aten_op
+    inplace_foreach_ops.add(aten_op)
+
+    def fn(*args, **kwargs):
+        results = outplace_op(*args, **kwargs)
+        mut_results = []
+        for arg, result in zip(args[0], results):
+            mut_results.append(mutate_to(arg, result, unsafe_alias=True))
```

Comment:
Do we need this? Why can't we just lower the `outplace_op` directly and express the mutation in `make_foreach_pointwise`?

Reply:
You would still need a lowering for the inplace version in this case though too right? I'm not sure what this would save tbh

Reply:
Morally speaking, I think `realize_into` is basically not needed here. I think of the entire point of `realize_into` as introducing a `copy_` op into the graph - in the case that we don't need to introduce another kernel, `realize_into` is just basically calling `realize()` on the `src` and `dst` and then modifying `src` to have a `MutationLayout`. So I feel like this is just obfuscating what's really going on 🤔 

cc: @oulgen on `MutationLayout` changes

Reply:
I see, yeah I can change this, I do still think having this lowering will make things clearer even if I remove the mutate_to call. In essence if I combine this with the outplace op, I will still need to know whether I'm the inplace op or not and need a mapping of outplace -> inplace ops in order to decide whether to apply the mutation or not, I feel like this lowering makes this clearer.

Reply:
Well actually also wdyt about what happens in `mutate_to`, it seems like there is some interesting logic related to views and stuff.